samsa2-master.nf

```diff
@@ -0,0 +1,258 @@
+    publishDir step_1_output_dir, mode: 'copy'
+    module 'trimmomatic'
+
+    input:
+    tuple val(sample_id), path(reads)
+
+    output:
+    tuple val("${sample_id}"), path("${sample_id}.cleaned.forward"), path("${sample_id}.cleaned.reverse")
+
+    script:
+    """
+    trimmomatic PE -phred33 -threads $task.cpus "$reads" ${sample_id}.cleaned.forward ${sample_id}.cleaned.forward_unpaired ${sample_id}.cleaned.reverse ${sample_id}.cleaned.reverse_unpaired SLIDINGWINDOW:4:15 MINLEN:70
+    """
+
+}
+
+// step 2
+process PEAR {
+    cpus 56
+    memory '56 GB'
+    publishDir step_2_output_dir, mode: 'copy'
+    module 'pear/0.9.11'
+
+    input:
+    tuple val(sample_id), path(reads_cleaned_forward), path(reads_cleaned_reverse)
+
+    output:
+    tuple val("$sample_id"), path("${sample_id}.merged.assembled.fastq"), path("${sample_id}.merged.discarded.fastq"), path("${sample_id}.merged.unassembled.forward.fastq"), path("${sample_id}.merged.unassembled.reverse.fastq"), path("${sample_id}.merged.assembled.fastq.ribosomes.log")
+
+    script:
+    """
+    pear -f $reads_cleaned_forward -r $reads_cleaned_reverse -j $task.cpus -o ${sample_id}.merged 2>&1 | tee ${sample_id}.merged.assembled.fastq.ribosomes.log
+    """
+}
+
+// step 2.9
+process RAWREADCOUNT {
+
+    cpus 1
+    memory '4 GB'
+    publishDir step_2_output_dir, mode: 'copy'
+
+    input:
+    path(input_forward_reads)
+
+    output:
+    path 'raw_counts.txt'
+
+    script:
+    """
+    for infile in $input_forward_reads; do python $params.python_scripts/raw_read_counter.py -I \$infile -O raw_counts.txt; done
+    """
+
+}
+
+process SORTMERNA {
+
+    cpus 24
+    memory '12 GB'
+    publishDir step_3_output_dir, mode: 'copy'
+
+    input:
+    tuple val(sample_id), path(assembled_fastq)
+
+    output:
+    tuple val(sample_id), path("${sample_id}.merged.ribodepleted.fastq")
+
+    script:
+    """
+    ${params.programs}/sortmerna-2.1/sortmerna \
+        -a $task.cpus \
+        --ref ${params.programs}/sortmerna-2.1/rRNA_databases/silva-bac-16s-id90.fasta,${params.programs}/sortmerna-2.1/index/silva-bac-16s-db \
+        --reads $assembled_fastq \
+        --aligned ${assembled_fastq}.ribosomes \
+        --other ${sample_id}.merged.ribodepleted \
+        --fastx \
+        --log \
+        -v
+    """
+}
+
+process DIAMOND_REFSEQ {
+
+    cpus 70
+    memory '170 GB'
+    publishDir "${step_4_output_dir}/RefSeq_results", mode: 'copy'
+
+    input:
+    tuple val(sample_id), path(ribodepleted_fastq)
+
+    output:
+    tuple val(sample_id), path("${sample_id}.merged.RefSeq_annotated"), path("${ribodepleted_fastq}.RefSeq.daa")
+
+    script:
+    """
+    ${params.programs}/diamond blastx --db $params.diamond_database -q $ribodepleted_fastq -a ${ribodepleted_fastq}.RefSeq -t /vast/scratch/users/\$USER/tmp -k 1 -p $task.cpus -b 12 -c 1
+    ${params.programs}/diamond view --daa ${ribodepleted_fastq}.RefSeq.daa -o ${sample_id}.merged.RefSeq_annotated -f tab -p $task.cpus
```

Comment:
I recommend using long names where possible, because it makes the code more self-documenting. e.g. `--query` instead of `-q`.

Reply:
opened issue https://github.com/WEHI-ResearchComputing/samsa2-nf/issues/3